src/jit.c

```diff
@@ -1252,6 +1232,147 @@ static void prepare_translate(struct jit_state *state)
 #endif
 }
 
+static int n_reg = ARRAYS_SIZE(register_map);
+static int count = 0;
+static int reg_table[32];
+static int vm_reg[3] = {0};
+
+static void reset_reg()
+{
+    count = 0;
+    for (int i = 0; i < 32; i++) {
+        reg_table[i] = -1;
+    }
+}
+
+static void store_back(struct jit_state *state)
+{
+    for (int i = 0; i < 32; i++) {
+        if (reg_table[i] != -1) {
+            emit_store(state, S32, reg_table[i], parameter_reg[0],
+                       offsetof(riscv_t, X) + 4 * i);
+        }
+    }
+}
+
+FORCE_INLINE void store_back_target(struct jit_state *state, int target_reg)
+{
+    for (int i = 0; i < 32; i++) {
+        if (reg_table[i] == target_reg) {
+            reg_table[i] = -1;
+            emit_store(state, S32, target_reg, parameter_reg[0],
+                       offsetof(riscv_t, X) + 4 * i);
```

Comment:
Can we do an early return here? The target should be unique in the `reg_table`.

Reply:
Yes, we should do early return here.